ddtrace/internal/logger.py

```diff
@@ -72,7 +77,7 @@ def hasHandlers(self):
         if not c.propagate:
             break
         else:
-            c = c.parent
+            c = c.parent  # type: ignore
```

Comment:
Is this due to `c.parent` being `None` at one point? Can we avoid ignoring here by adding a `type: Optional[...]` when `c` is first assigned to above?

Reply:
No, the error is somehow due to mypy confusing `c.parent` to be either a Logger or a PlaceHolder type, here's the actual error:
`Incompatible types in assignment (expression has type "Union[Logger, PlaceHolder]", variable has type "DDLogger") `